File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EEronprimaryDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EEronprimaryDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd17
            AppTheme.STANDARD -> R.color.cstd17
            AppTheme.BLUE -> R.color.cbd17
            AppTheme.GREEN -> R.color.cgd17
            AppTheme.RED -> R.color.crd17
            AppTheme.NOTHING -> R.color.cnod48
        }
    }
}

File: lib/layer2_core/dark/e_eronprimary_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EEronprimaryDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd17;
      case 'standard': return cstd17;
      case 'blue': return cbd17;
      case 'green': return cgd17;
      case 'red': return crd17;
      case 'nothing': return cnod48;
      default: return cstd17;
    }
  }
}
